From 0da910fb1bf2bd4982ca6541458c795ec296f398 Mon Sep 17 00:00:00 2001
From: Marc Fournier <marc.fournier@camptocamp.com>
Date: Fri, 8 Aug 2014 15:55:10 +0200
Subject: [PATCH 14/22] snmp: free snmp_pdu struct allocated by
 snmp_pdu_create()

This should fix the leak reported in issue #610.
---
 src/snmp.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/snmp.c b/src/snmp.c
index d440f7f..dacdede 100644
--- a/src/snmp.c
+++ b/src/snmp.c
@@ -1503,6 +1503,10 @@ static int csnmp_read_table (host_definition_t *host, data_definition_t *data)
     snmp_free_pdu (res);
   res = NULL;
 
+  if (req != NULL)
+    snmp_free_pdu (req);
+  req = NULL;
+
   if (status == 0)
     csnmp_dispatch_table (host, data, instance_list_head, value_list_head);
 
-- 
1.9.3

